Finish expanding of AWS DDB tests #3114 (#3116)
apache/camel-quarkus: extensions/aws2-ddb/runtime/src/doc/configuration.adoc

The aws2-ddb extension is using a DynamoDB client under the hood.

> The underlying DynamoDB client could either be created by this extension itself or by the `quarkus-amazon-dynamobdb`.

Reviewer: ```suggestion
The underlying DynamoDB client could either be created by this extension itself or by the `quarkus-amazon-dynamodb` from Quarkus project.
```

=== Client created by this extension

In that case fill following properties:

[source,properties]
----
camel.component.aws2-ddb.access-key=${AWS_ACCESS_KEY}
camel.component.aws2-ddb.secret-key=${AWS_SECRET_KEY}
camel.component.aws2-ddb.region=${AWS_REGION:us-east-1}
----

or

[source,properties]
----
camel.component.aws2-ddbstream.access-key=${AWS_ACCESS_KEY}
camel.component.aws2-ddbstream.secret-key=${AWS_SECRET_KEY}
camel.component.aws2-ddbstream.region=${AWS_REGION:us-east-1}
----

> === Client created by `quarkus-amazon-dynamodb`

Reviewer: ```suggestion
=== Client created by `quarkus-amazon-dynamodb` extension
```

> To use client created by `quarkus-amazon-dynamodb` extension, follow the  https://quarkus.io/guides/amazon-dynamodb#configuring-dynamodb-clients[documentation] with some restrictions:

Reviewer: ```suggestion
To use the client created by `quarkus-amazon-dynamodb` extension, follow the  https://quarkus.io/guides/amazon-dynamodb#configuring-dynamodb-clients[documentation] with some restrictions:
```

> - Apache client type has to be selected

Reviewer: ```suggestion
- Apache client type has to be selected
+
```

A single `+` sign on a line makes following code block belong to the previous list item

[source,properties]
----
quarkus.dynamodb.sync-client.type=apache
----

> 

Reviewer: ```suggestion
+
```

> - All required dependencies are already part of the extension.
> There is no need to add them.

Reviewer: It looks like the opposite is true, isn't it? 
```suggestion
- You need to add the `io.quarkus:quarkus-amazon-dynamodb` to your application.
```

> 

Reviewer: ```suggestion
```

> - `DynamoDbClient` has to be injected somewhere in the application - this triggers creation of client via `quarkus-amazon-dynamodb`:

Reviewer 2: Wouldn't it be autowired? When camel does the lookup of `DynamoDbClient` against ARC, that should trigger creation of the bean (assuming the relevant config properties were provided).

Author: Unfortunatelly quarkus extension is looking for injection points, see https://github.com/quarkusio/quarkus/blob/main/extensions/amazon-services/common/deployment/src/main/java/io/quarkus/amazon/common/deployment/AbstractAmazonServiceProcessor.java#L64

Reviewer 3: So it there any room to improve in Quarkus ?  Add an opiton to force create the client ?

> There are 2 ways of DynamoDB client creation:

Reviewer 4: What would think about the wording below ? If I get it right at all :)
The aws2-ddb extension is using a DynamoDB client under the hood. The underlying DynamoDB client could either be created by this extension itself or by the quarkus-amazon-dynamobdb.

Author: Nice:) this explains the reality much better, I'll use it

Author: done

=== Client created by this extension

In that case fill following properties:

[source,properties]
----
camel.component.aws2-ddb.access-key=${AWS_ACCESS_KEY}
camel.component.aws2-ddb.secret-key=${AWS_SECRET_KEY}
camel.component.aws2-ddb.region=${AWS_REGION:us-east-1}
----

or

[source,properties]
----
camel.component.aws2-ddbstream.access-key=${AWS_ACCESS_KEY}
camel.component.aws2-ddbstream.secret-key=${AWS_SECRET_KEY}
camel.component.aws2-ddbstream.region=${AWS_REGION:us-east-1}
----

=== Client created by `quarkus-amazon-dynamodb`

o use client created by `quarkus-amazon-dynamodb` extension, follow the  https://quarkus.io/guides/amazon-dynamodb#configuring-dynamodb-clients[documentation] with some restrictions:

- Apache client type has to be selected
[source,properties]
----
quarkus.dynamodb.sync-client.type=apache
----

> - All required dependencies are already part of the extension. There is no need to add them.

Reviewer 4: If there is no need to add them, maybe we don't show the example below ?

Author: That makes sense, I'll remove it immediately.

Author: done
